src/js/utils/process-json.js

```diff
@@ -83,13 +97,21 @@ function processDelinquencies( datasets ) {
 }
 
 /**
- * Returns a data object with data starting in January 2009 for use in all line charts
+ * Returns data starting in January 2009 for use in all line charts.
  *
- * @param {number} data - response from requested JSON file
- * @param {string} group - optional parameter for specifying if the chart requires use of a "group" property in the JSON, for example the charts with a group of "Younger than 30" will filter data to only include values matching that group
- * @returns {Object} data - object with adjusted and unadjusted value arrays containing timestamps and a number value
+ * @param {number} data - response from requested JSON file.
+ * @param {string} [group] -
+ *   Optional parameter for specifying if the chart requires use of a "group"
+ *   property in the JSON, for example the charts with a group of "Younger
+ *   than 30" will filter data to only include values matching that group.
+ * @param {string} [source] -
+ *   Optional parameter for the file url and name. Used for inquiry index files
+ *   which have 4 months of projected data instead of 6.
+ * @returns {Object} data -
+ *   Object with adjusted and unadjusted value arrays containing timestamps
+ *   and a number value.
  */
-function processNumOriginationsData( data, group ) {
+function processNumOriginationsData( data, group, source ) {
```

Comment:
Should the indices actually have their own function? We're passing in the `source` here for for inquiries and denials but this function implies its for originations data.

Reply:
this function is for Number of Originations and Volume of Originations charts, plus Inquiry index and Inferred Denials. I'm reluctant to add another function and duplicate the code that's reusable. I'm thinking we should rename it to be more accurate, at the very least for now?

There are all the processing functions-
processDelinquencies,
processNumOriginationsData,
processYoyData,
processMapData,

Reply:
We can use a private function for the shared parts. We have a lot of processing functions, but that seems accurate since there are a lot of formats to the data, right? The function should do one thing well and right now the `metadata` and `source` arguments are both used in different contexts within the same function. Unless we foresee inquiry/denials using the `group` (what `metadata` becomes), that doesn't seem to make sense. Opened issue https://github.cfpb.gov/CFGOV/platform/issues/3012